src/libs/ajax/DatasetBuilder.ts

```diff
@@ -69,23 +95,88 @@ export interface GetConceptsResponse {
 type DatasetRequest = {
   cohorts: Cohort[];
   conceptSets: ConceptSet[];
-  valuesSets: { domain: string; values: DatasetBuilderValue[] }[];
+  valueSets: { domain: string; values: DatasetBuilderValue[] }[];
+};
+
+export type DatasetRequestAPI = {
+  cohorts: CohortAPI[];
+  conceptSets: ConceptSet[];
+  valueSets: { id: number; name: string; values: string[] }[];
 };
 
-type DatasetAccessRequest = {
+export type DatasetAccessRequest = {
   name: string;
   researchPurposeStatement: string;
   datasetRequest: DatasetRequest;
 };
 
+export type DatasetAccessRequestAPI = {
+  name: string;
+  researchPurposeStatement: string;
+  datasetRequest: DatasetRequestAPI;
+};
+
+export const convertValueSets = (valueSets: { domain: string; values: DatasetBuilderValue[] }[]) => {
+  return _.map(
+    (valueSet) => ({ id: 0, name: valueSet.domain, values: _.map(({ name }) => name, valueSets.values) }),
+    valueSets
+  );
+};
+
+export const convertCohorts = (cohorts: Cohort[]) => {
+  return _.map(
+    (cohort) => ({
+      name: cohort.name,
+      criteriaGroups: _.map(
+        (criteriaGroup) => ({
+          name: criteriaGroup.name,
+          mustMeet: criteriaGroup.mustMeet,
+          meetAll: criteriaGroup.meetAll,
+          count: criteriaGroup.count,
+          criteria: _.map((criteria) => convertCriteria(criteria), criteriaGroup.criteria),
+        }),
+        cohort.criteriaGroups
+      ),
+    }),
+    cohorts
+  );
+};
+
+export const convertCriteria = (criteria: AnyCriteria) => {
+  if (criteria.kind === 'range') {
+    return { kind: criteria.kind, name: criteria.name, id: criteria.id, low: criteria.low, high: criteria.high };
+  }
+  if (criteria.kind === 'list') {
+    return {
+      kind: criteria.kind,
+      name: criteria.name,
+      id: criteria.id,
+      values: _.map((value) => value.id, criteria.values),
+    };
+  }
+  return { kind: criteria.kind, name: criteria.name, id: criteria.id };
```

Comment:
I would use a `switch` here, although maybe there's a better way than that. With `switch` I would also add a case for unknown `kind` instead of assuming the default case is `domain`, in case the types changes in the future.

Reply:
I'd throw an error rather than `return criteria as never`. I think that will be louder and allow you to put a custom message.

Reply:
For the way to do things, there is a better way! You could do:
`_.merge({kind: criteria.kind, name: criteria.name, id: criteria.id}, ...(switch`

That would enable you to pull the shared types out and use destructuring to get all the other values you need.